### PR TITLE
SDIT-2192 Add endpoint to retrieve splash screen config

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.CodeDescription
+
+@RestController
+@Validated
+@RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
+class SplashScreenResource(private val service: SplashScreenService) {
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_SCREEN_ACCESS')")
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt:22]: Wasn't sure of a role name to use (there may be a  more general one for admin?)

File: src/main/resources/db/migration/V1_1101__SPLASH_SCREEN_FUNCS.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
+create table SPLASH_SCREEN_FUNCS
```
Comment (karenmillermoj) [inline src/main/resources/db/migration/V1_1101__SPLASH_SCREEN_FUNCS.sql:1]: This table is not used in Nomis but the table I am interested in depends on it.
The table allows pl/sql functions to be added - but currently this is not done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt (hunk 1 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt:22]: Difficult ones for these end points that cross domains. So maybe stick with this

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**